Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,70 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
-        puts
-        puts "External commands"
-        puts Formatter.columns(exts)
+    # TODO: Goal is to not require this function at all.
+    command = CommandsCommand.new
+    command.check_for_errors
```

Comment:
I'm not sure I follow. Is the reason for the failure a deficiency in testing, or is because `check_for_errors` is using some instance state that isn't available yet?

If it's the former, we can flag it to be fixed later. If the latter, then we should make sure that the state `check_for_errors` requires is already available by the time it gets called in `initialize`.

Basically, we don't want to separate initialization from error checking, since it doesn't make sense to initialize and return an object that we immediately have to test explicitly (and then discard).